core/src/main/java/org/geysermc/geyser/translator/inventory/InventoryTranslator.java

```diff
@@ -216,6 +220,22 @@ public ItemStackResponse translateRequest(GeyserSession session, Inventory inven
                     boolean isSourceCursor = isCursor(transferAction.getSource());
                     boolean isDestCursor = isCursor(transferAction.getDestination());
 
+                    if (destSlot == 5 || sourceSlot == 5) {
+                        if (destSlot == 5) {
+                            //only set the head if the destination is the head slot
+                            GeyserItemStack javaItem = inventory.getItem(sourceSlot);
+                            ItemData itemData = javaItem.getItemData(session);
+                            if (javaItem.getJavaId() == Items.PLAYER_HEAD.javaId()
```

Comment:
Can be `javaItem.asItem() == Items.PLAYER_HEAD`.

Reply:
sure, fixed in new commit